Document that testable representations of all pipeline contexts is provided

andreasohlund opened the pull request.

bording commented on nservicebus/pipeline/index.md:
```diff
@@ -12,18 +12,20 @@ related:
- * [DataBus](/nservicebus/messaging/databus/)
- * [Message Property Encryption](/nservicebus/security/property-encryption.md)
- * [Recoverability](/nservicebus/recoverability/)
+## Features build on the pipeline
```
Should be "built" not "build"

bording commented on nservicebus/testing/index_content_testing_[6,).partial.md:
```diff
@@ -29,4 +26,6 @@ snippet: SampleBehavior
\ No newline at end of file
+snippet: BehaviorTest
+
+Testable representations are provided for all pipeline behaviour contexts, see [the pipeline documentation](/nservicebus/pipeline/) for further details.
```
Should be "behavior", not "behaviour".

Comma should be changed to a period, "see" should be "See"

timbussmann commented on nservicebus/testing/index_content_testing_[6,).partial.md:
```diff
@@ -20,7 +18,6 @@ The following unit tests checks that the total amount has the discount applied:
 snippet: SagaTest
 
-
 ### Testing a behavior
```
should this be moved one level up too then?

andreasohlund commented:
@timbussmann @bording addressed your comments.

Thoughts on perhaps adding a bullet list of all the testable context and what they mock to improve the searchability [mentioned by @SimonCropp ](https://github.com/Particular/docs.particular.net/issues/3971#issuecomment-413355797)?

andreasohlund commented:
In that case lets get this PR in. We can always add the list later on
